src/Storages/S3Queue/StorageS3Queue.cpp

```diff
@@ -106,42 +105,23 @@ StorageS3Queue::StorageS3Queue(
     if (!withGlobs())
         throw Exception(ErrorCodes::QUERY_NOT_ALLOWED, "S3Queue url must either end with '/' or contain globs");
 
-    String setting_zk_path = s3queue_settings->keeper_path;
-    if (setting_zk_path.empty())
-    {
-        auto database = DatabaseCatalog::instance().getDatabase(table_id_.database_name);
-        bool is_in_replicated_database = database->getEngineName() == "Replicated";
-
-        auto default_path = getContext()->getSettingsRef().s3queue_default_zookeeper_path.value;
-        String zk_path_prefix;
-
-        if (!default_path.empty())
-        {
-            zk_path_prefix = default_path;
-        }
-        else if (is_in_replicated_database)
-        {
-            LOG_INFO(log, "S3Queue engine zookeeper path is not specified. "
-                     "Using replicated database zookeeper path");
-
-            zk_path_prefix = fs::path(assert_cast<const DatabaseReplicated *>(database.get())->getZooKeeperPath()) / "s3queue";
-        }
-        else
-        {
-            throw Exception(ErrorCodes::NO_ZOOKEEPER,
-                            "S3Queue keeper_path engine setting not specified, "
-                            "s3queue_default_zookeeper_path_prefix not specified");
-        }
+    std::string zk_path_prefix = getContext()->getSettingsRef().s3queue_default_zookeeper_path.value;
+    if (zk_path_prefix.empty())
+        zk_path_prefix = "/";
 
-        zk_path = zkutil::extractZooKeeperPath(
-            fs::path(zk_path_prefix) / toString(table_id_.uuid), /* check_starts_with_slash */ true, log);
+    std::string result_zk_path;
+    if (s3queue_settings->keeper_path.changed)
+    {
+        /// We do not add table uuid here on purpose.
```

Comment:
Ok. Just in case - this should not lead to path traversal attacks: if someone adds `..`, the path is not normalized.